tokio/src/runtime/task/harness.rs

```diff
@@ -473,13 +477,20 @@ fn poll_future<T: Future>(core: &CoreStage<T>, cx: Context<'_>) -> Poll<()> {
     let output = match output {
         Ok(Poll::Pending) => return Poll::Pending,
         Ok(Poll::Ready(output)) => Ok(output),
-        Err(panic) => Err(JoinError::panic(panic)),
+        Err(panic) => {
+            scheduler.unhandled_panic();
```

Comment:
It's only unhandled if the `JoinHandle` gets dropped, no?

Reply:
"unhandled" refers to unhandled by the spawn task. As per #4516, two separate behaviors are to always shutdown if the task panics or to only shutdown if the JoinHandle is dropped. I implemented the first one here.

Reply:
FWIW, I think this option will be most useful for **tests** where the application is expected to gracefully handle all possible user inputs.